src/actions/websocket.ts

```diff
@@ -576,7 +579,33 @@ function handleChannelDeletedEvent(msg: WebSocketMessage) {
                 EventEmitter.emit(General.DEFAULT_CHANNEL, '');
             }
 
-            dispatch({type: ChannelTypes.RECEIVED_CHANNEL_DELETED, data: {id: msg.data.channel_id, team_id: msg.data.team_id, deleteAt: msg.data.delete_at, viewArchivedChannels}}, getState);
+            dispatch({type: ChannelTypes.RECEIVED_CHANNEL_DELETED, data: {id: msg.data.channel_id, team_id: msg.data.team_id, viewArchivedChannels}}, getState);
+
+            dispatch(fetchMyChannelsAndMembers(currentTeamId));
+        }
+        return {data: true};
+    };
+}
+
+function handleChannelUnarchiveEvent(msg: WebSocketMessage) {
+    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState();
+        const currentChannelId = getCurrentChannelId(state);
+        const currentTeamId = getCurrentTeamId(state);
+        const config = getConfig(state);
+        const viewArchivedChannels = config.ExperimentalViewArchivedChannels === 'true';
+
+        if (msg.broadcast.team_id === currentTeamId) {
+            if (msg.data.channel_id === currentChannelId && !viewArchivedChannels) {
+                const channelsInTeam = getChannelsNameMapInTeam(state, currentTeamId);
+                const channel = getChannelByName(channelsInTeam, getRedirectChannelNameForTeam(state, currentTeamId));
+                if (channel && channel.id) {
+                    dispatch({type: ChannelTypes.SELECT_CHANNEL, data: channel.id});
```

Comment:
This seems like it'll make everyone switch to a channel that's unarchived by another user. Like with the action, perhaps this should be triggered explicitly by whatever component attempts to unarchive the channel?